### PR TITLE
feat(python): Render string columns in notebooks without quotes

### PR DESCRIPTION
**Pull request by stinodego**
Closes https://github.com/pola-rs/polars/issues/6510

This is consistent with the regular DataFrame repr.

I've looked into implementing a setting to toggle this on/off but it turned out to be more difficult than I initially anticipated :sweat_smile:

### CODE REVIEW

File: py-polars/polars/config.py [hunk 1]
```diff
@@ -686,10 +686,6 @@ def set_fmt_str_lengths(cls, n: int | None) -> type[Config]:
         if n is None:
             os.environ.pop("POLARS_FMT_STR_LEN", None)
         else:
-            if n <= 0:
-                msg = "number of characters must be > 0"
-                raise ValueError(msg)
```
Comment (stinodego) [inline py-polars/polars/config.py:691]: Negative limit is now parsed as 'no limit' and 0 is a valid limit.

**Comment by ritchie46** [comment]
I am not a fan of this. I'd rather also show the quotes in the `DataFrame` repr. It helps so much with detecting whitespace.

**Comment by ritchie46** [comment]
I am not a fan of this. I'd rather also show the quotes in the `DataFrame` repr. It helps so much with detecting whitespace.

**Comment by stinodego** [comment]
In that case, I'll try to finish the PR for a setting to toggle the quotes on and off. We can default to "on" then. I just have to figure out how to make it work in the pretty table repr.

**Pull request closed without merging**